internal/cmd/gocdk/biome.go

```diff
@@ -246,17 +263,98 @@ func biomeApply(ctx context.Context, pctx *processContext, biome string, input b
 		return xerrors.Errorf("biome apply %s: %w", biome, err)
 	}
 
-	c := pctx.NewCommand(ctx, biomePath, "terraform", "init", "-input="+strconv.FormatBool(input))
-	if err := c.Run(); err != nil {
+	writeTerraformOutput := func(cmd *exec.Cmd, msg []byte) {
+		const separator = "#########################################\n"
+		pctx.stderr.Write([]byte("\n"))
+		pctx.stderr.Write([]byte(separator))
+		pctx.stderr.Write([]byte(fmt.Sprintf("Output from %q\n", strings.Join(cmd.Args, " "))))
+		pctx.stderr.Write([]byte(separator))
+		pctx.stderr.Write(msg)
+		pctx.stderr.Write([]byte(separator))
+		pctx.stderr.Write([]byte("\n"))
+	}
+
+	pctx.Logf("Checking to see if resource updates are needed...")
+
+	// First, run "terraform init". It's needed the first time, and anytime
+	// a new Terraform provider is added, so it's safest to just always run it.
+	inputArg := fmt.Sprintf("-input=%v", opts.Input)
+	c := pctx.NewCommand(ctx, biomePath, "terraform", "init", inputArg)
+	c.Env = overrideEnv(c.Env, "TF_IN_AUTOMATION=1")
+	c.Stdout, c.Stderr = nil, nil
+	if out, err := c.CombinedOutput(); err != nil {
+		writeTerraformOutput(c, out)
+		return xerrors.Errorf("biome apply %s: %w", biome, err)
+	} else if opts.Verbose {
+		writeTerraformOutput(c, out)
+	}
+
+	// Next, run "terraform plan", writing the plan to a temp file.
+	planFile, err := ioutil.TempFile(pctx.workdir, "tfplan")
+	if err != nil {
+		return xerrors.Errorf("biome apply %s: %w", biome, err)
+	}
+	// Note: using just the base filename instead of passing the file path to --out
+	// because Terraform appears to not work with full Windows paths.
+	planFileName := filepath.Base(planFile.Name())
+	defer planFile.Close()
+	defer os.Remove(planFile.Name())
+	c = pctx.NewCommand(ctx, biomePath, "terraform", "plan", "-detailed-exitcode", "-out="+planFileName, inputArg)
+	c.Env = overrideEnv(c.Env, "TF_IN_AUTOMATION=1")
+	c.Stdout, c.Stderr = nil, nil
+
+	// No error means success and no diffs.
+	// Exit code 1 means the command failed.
+	// Exit code 2 means the command succeeded, but there's a diff.
+	if out, err := c.CombinedOutput(); err != nil && !strings.Contains(err.Error(), "exit status 2") {
```

Comment:
Using a `strings.Contains` for this is brittle. Instead, use `xerrors.As` to extract the [`*exec.ExitError`](https://godoc.org/os/exec#ExitError) and use [`os.ProcessState.ExitCode()`](https://godoc.org/os#ProcessState.ExitCode) to test for this condition.

Reply:
Ooh, that would be great... but it doesn't work in Go 1.11. I can leave a `TODO` to fix it after 1.11 support is dropped...?